### DISCUSSION TITLE
Using Saleor with third party inventory api

### DISCUSSION THREAD

**Discussion by thecodingtree** [saleor/saleor#9011] [Q&A]
I am currently considering using Saleor for a parts website, I am very impressed with the platform but I have a question regarding integrating a third party inventory management api with the Saleor platform.

Our client uses an existing inventory management system called Lightspeed which has an API to retrieving parts info, processing orders, etc... 

So my question is, what would be the best way to integrate Lightspeed into Saleor in order to pull parts info? My current thought is to either create some type of product sync task which periodically imports the parts info from Lightspeed and creates/updates Saleor Products. Or create a proxy which goes out to Lightspeed in real time to fetch the parts info. 

I don't know enough about the platform yet to make a decision on which path to go down which is why I am reaching out to this community for help.  

Thanks!

**Comment by karolkielecki** [comment]
Hi @thecodingtree, 

If your system contains webhooks, you can create an app. Learn more here: https://docs.saleor.io/docs/3.0/developer/extending/apps/key-concepts

**Reply by thecodingtree** [reply]
Thanks @karolkielecki, 

Unfortunately the third party system we want to integrate with does not contain webhooks. Our interface is just a REST API with all of the product info. I am leaning towards creating a sync task (or multiple tasks) to pull in the product data from the third party API and create copies of the products locally in our Saleor backend. Just wondering if this seems like the right path forward or if there is a better way of doing this. I feel like this is a common use case for Saleor but haven't found any good solutions.

**Reply by darrylmorley** [reply]
@thecodingtree I'm interested in where you've got to with this. We too are using Lightspeed Retail as an in-store EPOS and currently running 2 websites off the API with a Postgres DB in the middle (I'm just using cron to run a couple of scripts to keep the DB in sync with Lightspeed). I'm considering replacing the DB with Saleor to provide greater flexibility over the products in terms of categorization etc. Is this a suitable use case I wonder?

**Reply by korycins** [reply]
Hey @thecodingtree  @darrylmorley 

I think that you could build a Saleor App and mount it between Saleor and Lightspeed. The app could use a Saleor's webhook (if there is a need) to sync in direction: Saleor -> Lightspeed (where App could map Saleor payload into Lightspeed payload). 
Your app could also have some scheduled tasks to fetch data from Lightspeed and sync it with Saleor. 
Additionally, your app could mount an iframe page into the Saleor Dashboard (So for example you can manage your migrations from Saleor Dashboard directly).

App docs: https://docs.saleor.io/docs/3.0/developer/extending/apps/key-concepts
Our framework for building an apps. You can find there an app samples: https://github.com/saleor/saleor-app-framework-python

The one thing that I believe is missing in docs is that when you're having an app manifest you can install it directly from dashboard:
http://<dashboard-url>/apps/install?manifestUrl=https://<app-domain>/configuration/manifest

**Reply by darrylmorley** [reply]
@korycins Thank you for the pointers , I'm going to look further into this.
